lib/spack/spack/spec.py

```diff
@@ -188,8 +188,10 @@
                  '^': dependency_color,
                  '#': hash_color}
 
-"""Regex used for splitting by spec field separators."""
-_separators = '[%s]' % ''.join(color_formats.keys())
+"""Regex used for splitting by spec field separators.
+These need to be escaped to avoid metacharacters in color_formats.keys().
+"""
```

Comment:
I looked at it again and putting variable docstrings after variables makes me twitchy.  So let's go with google-style everywhere except variables, and `#:` for variables.